src/mkdocstrings/handlers/python.py

```diff
@@ -65,15 +66,14 @@ class PythonRenderer(BaseRenderer):
     """  # noqa: E501
 
     def render(self, data: Any, config: dict) -> str:  # noqa: D102 (ignore missing docstring)
-        final_config = dict(self.default_config)
-        final_config.update(config)
+        final_config = ChainMap(config, self.default_config)
 
         template = self.env.get_template(f"{data['category']}.html")
 
         # Heading level is a "state" variable, that will change at each step
         # of the rendering recursion. Therefore, it's easier to use it as a plain value
         # than as an item in a dictionary.
-        heading_level = final_config.pop("heading_level")
+        heading_level = final_config["heading_level"]
```

Comment:
I guess we can't pop from a `ChainMap`? Anyway, I also think it doesn't hurt to let `heading_level` in the final config, as the original value :+1:

Reply:
You can technically `pop`, but really it gets forwarded to the first dict, which is a totally different operation. E.g. if both dicts have the key, then it'd just make the value of the 2nd one visible instead...

Reply:
Okay, thanks for the explanation, LGTM.